### ISSUE TITLE
cv.selbal: Error in xtfrm.data.frame(x) : cannot xtfrm data frames

### ISSUE THREAD

**Issue by mariamaseng** [malucalle/selbal#32]
Hi!
I am trying this amazing package for the first time, and are encountering a problem. It says it finished, defining the optimal number of variables as 6, but trying to access the object (CV.BAL.dic) it is not found. I am getting an error message I dont understand, see below. 

This is the code: 
```
x = final_selbal_df[,1:150] #dim(x) 667 150
y = final_selbal_df[,151] #length(y) 667
CV.BAL.dic = selbal.cv(x, y, n.fold = 5, n.iter = 10,logit.acc = "AUC")
```

And this is the output:
############################################################### 
 STARTING selbal.cv FUNCTION 
###############################################################

#-------------------------------------------------------------# 
# ZERO REPLACEMENT . . .

Warning: Column(s) containing more than 80% zeros/unobserved values were found (check it out using zPatterns).
                    (You can use the z.warning argument to modify the warning threshold).Warning: Row(s) containing more than 80% zeros/unobserved values were found (check it out using zPatterns).
                  (You can use the z.warning argument to modify the warning threshold).
, . . . FINISHED. 
#-------------------------------------------------------------#

#-------------------------------------------------------------# 
# Starting the cross - validation procedure . . .Warning: already exporting variable(s): logit.acc
 . . . finished. 
#-------------------------------------------------------------# 
###############################################################

 The optimal number of variables is: 6 

Warning: Column(s) containing more than 80% zeros/unobserved values were found (check it out using zPatterns).
                    (You can use the z.warning argument to modify the warning threshold).Warning: Row(s) containing more than 80% zeros/unobserved values were found (check it out using zPatterns).
                  (You can use the z.warning argument to modify the warning threshold).Warning: Ignoring unknown parameters: `linewidth`Warning: The `<scale>` argument of `guides()` cannot be `FALSE`. Use "none" instead as of ggplot2 3.3.4.Error in xtfrm.data.frame(x) : cannot xtfrm data frames

**Comment by mariamaseng** [comment]
Additional messages: 


<html>
<body>
<div class="GND-IWGDDID ace_constant" style="display: inline; overflow-wrap: break-word; word-break: break-word; color: rgb(197, 6, 11); font-family: &quot;Lucida Console&quot;, monospace; font-size: 13.3333px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: pre-line; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span>Error in xtfrm.data.frame(x) : cannot xtfrm data frames</span></div><span style="color: rgb(0, 0, 0); font-family: &quot;Lucida Console&quot;, monospace; font-size: 13.3333px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: pre-line; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(230, 230, 230); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><div class="GND-IWGDEID stack_trace" aria-hidden="true" style="margin-top: 0.5em; margin-left: 0.3em; color: rgb(0, 0, 0); font-family: &quot;Lucida Console&quot;, monospace; font-size: 13.3333px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: pre-line; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: block;">

10. | stop("cannot xtfrm data frames")
-- | --


</div>
</body>
</html>

**Comment by peter0k** [comment]
> Additional messages:
> 
> ![image](https://user-images.githubusercontent.com/66627015/257828033-47d99683-b4da-44fd-a8e7-243fcffd8153.png)

Hi! I am also a first time user and having the same issue (R version 4.3.1).

**Comment by peter0k** [comment]
> Additional messages:
> 
> ![image](https://user-images.githubusercontent.com/66627015/257828033-47d99683-b4da-44fd-a8e7-243fcffd8153.png)

Modified 

    RESUME.BAL <- RESUME.BAL[,c(ncol(RESUME.BAL), ncol(RESUME.BAL)-1,
                                order(RESUME.BAL[nrow(RESUME.BAL),
                                                 -c(ncol(RESUME.BAL),
                                                    ncol(RESUME.BAL)-1)],
                                      decreasing = T))]
to

    RESUME.BAL <- RESUME.BAL[,c(ncol(RESUME.BAL), ncol(RESUME.BAL)-1,
                                order(as.numeric(RESUME.BAL[nrow(RESUME.BAL),
                                                 -c(ncol(RESUME.BAL),
                                                    ncol(RESUME.BAL)-1)],
                                      decreasing = T)))]

**Comment by peter0k** [comment]
> Hi @peter0k and @mariamaseng!
> 
> Did you solve the issue with `Error in xtfrm.data.frame(x) : cannot xtfrm data frames`?

Yes I sourced Selbal_Functions.R with the modification I posted above into my script.
It didn't produce the error  mentioned.
